<a href="https://colab.research.google.com/github/HansikaGunasekara/travel-insurance-claims-analysis/blob/main/Analyse_travel_insurance_data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [23]:
import pandas as pd

In [24]:
# Load & Preview data
df = pd.read_csv('travel_insurance.csv')
df.head()


Agency    Agency Type Distribution Channel                     Product Name  \
0    CBH  Travel Agency              Offline               Comprehensive Plan   
1    CBH  Travel Agency              Offline               Comprehensive Plan   
2    CWT  Travel Agency               Online  Rental Vehicle Excess Insurance   
3    CWT  Travel Agency               Online  Rental Vehicle Excess Insurance   
4    CWT  Travel Agency               Online  Rental Vehicle Excess Insurance   

  Claim  Duration Destination  Net Sales  Commision (in value) Gender  Age  
0    No       186    MALAYSIA      -29.0                  9.57      F   81  
1    No       186    MALAYSIA      -29.0                  9.57      F   71  
2    No        65   AUSTRALIA      -49.5                 29.70    NaN   32  
3    No        60   AUSTRALIA      -39.6                 23.76    NaN   32  
4    No        79       ITALY      -19.8                 11.88    NaN   41

In [25]:
# Check dimensions
df.shape

(63326, 11)

In [26]:
# List columns
list(df)

['Agency',
 'Agency Type',
 'Distribution Channel',
 'Product Name',
 'Claim',
 'Duration',
 'Destination',
 'Net Sales',
 'Commision (in value)',
 'Gender',
 'Age']

In [27]:
# Count claims
df.shape[0]

63326

In [28]:
# Unique product names
list(df['Product Name'].unique())

['Comprehensive Plan',
 'Rental Vehicle Excess Insurance',
 'Value Plan',
 'Basic Plan',
 'Premier Plan',
 '2 way Comprehensive Plan',
 'Bronze Plan',
 'Silver Plan',
 'Annual Silver Plan',
 'Cancellation Plan',
 '1 way Comprehensive Plan',
 'Ticket Protector',
 '24 Protect',
 'Gold Plan',
 'Annual Gold Plan',
 'Single Trip Travel Protect Silver',
 'Individual Comprehensive Plan',
 'Spouse or Parents Comprehensive Plan',
 'Annual Travel Protect Silver',
 'Single Trip Travel Protect Platinum',
 'Annual Travel Protect Gold',
 'Single Trip Travel Protect Gold',
 'Annual Travel Protect Platinum',
 'Child Comprehensive Plan',
 'Travel Cruise Protect',
 'Travel Cruise Protect Family']

In [29]:
#  Get the number of accepted & rejected claims
df['Claim'].value_counts()

Claim
No     62399
Yes      927
Name: count, dtype: int64

In [30]:
# Derive claim acceptance ratio
# claim acceptance_ratio = number of accepted claims / number of claims (%)
(df[df['Claim'] == 'Yes']['Claim'].value_counts() / df.shape[0]) * 100

Claim
Yes    1.463854
Name: count, dtype: float64

In [31]:
# Find the top destination for majority of claims
counts_per_destination = list(df['Destination'].value_counts())
countries_list = df['Destination'].value_counts().keys()
ind = counts_per_destination.index(max(counts_per_destination))
countries_list[ind]

'SINGAPORE'

In [32]:
# Find the top ranked destination for the rejected claims
counts_per_destination = list(df[df['Claim'] == 'No'].groupby('Destination')['Claim'].value_counts())
countries_list = df[df['Claim'] == 'No'].groupby('Destination')['Claim'].value_counts().keys()
counts_per_destination.index(max(counts_per_destination))
countries_list[118]

('SINGAPORE', 'No')

In [33]:
# Claims per agency
df['Agency'].value_counts()

Agency
EPX    35119
CWT     8580
C2B     8267
JZI     6329
SSI     1056
JWT      749
RAB      725
LWC      689
TST      528
KML      392
ART      331
CCR      194
CBH      101
TTW       98
CSR       86
ADM       82
Name: count, dtype: int64

In [34]:
# Rejection counts per agency
df[df['Claim'] == 'No']['Agency'].value_counts()

Agency
EPX    34924
CWT     8494
C2B     7720
JZI     6298
SSI     1049
JWT      745
RAB      724
LWC      653
TST      526
KML      384
ART      330
CCR      191
CBH      100
TTW       94
CSR       85
ADM       82
Name: count, dtype: int64

#**📊 Claims Analysis**
### Based on the number of claims made, what does this tell us about customer behaviour?

In the dataset out of 63K+ instances 927 instances are accepted. The claim acceptance raio is <2%.

###Should the insurance company be concerned?

Yes the data shows imbalance with a majority of rejections.

The destination for majority of the submitted claims & accepted claims is Singapore which is a concern.

The 'EPX' agency accounts for ~35K rejections. 'C2B', 'CWT' & 'JZI' accounts for 6K+ rejections.



In [35]:
# Frequency of claims for different products
df['Product Name'].value_counts()

Product Name
Cancellation Plan                       18630
2 way Comprehensive Plan                13158
Rental Vehicle Excess Insurance          8580
Basic Plan                               5469
Bronze Plan                              4049
1 way Comprehensive Plan                 3331
Value Plan                               2715
Silver Plan                              2249
Annual Silver Plan                       1423
Ticket Protector                         1056
Travel Cruise Protect                     527
Comprehensive Plan                        364
Gold Plan                                 352
24 Protect                                247
Single Trip Travel Protect Gold           204
Premier Plan                              194
Annual Gold Plan                          194
Single Trip Travel Protect Silver         173
Annual Travel Protect Gold                100
Annual Travel Protect Silver               86
Individual Comprehensive Plan              74
Single Trip Travel Protect Platinum        73
Annual Travel Protect Platinum             53
Spouse or Parents Comprehensive Plan       15
Child Comprehensive Plan                    9
Travel Cruise Protect Family                1
Name: count, dtype: int64

In [36]:
# Frequency of products grouped into agency type
df.groupby('Agency Type')['Product Name'].value_counts()

Agency Type    Product Name                        
Airlines       Basic Plan                               5469
               Bronze Plan                              4049
               Value Plan                               2412
               Silver Plan                              2249
               Annual Silver Plan                       1423
               Ticket Protector                         1056
               Gold Plan                                 352
               24 Protect                                247
               Annual Gold Plan                          194
               Premier Plan                                6
Travel Agency  Cancellation Plan                       18630
               2 way Comprehensive Plan                13158
               Rental Vehicle Excess Insurance          8580
               1 way Comprehensive Plan                 3331
               Travel Cruise Protect                     527
               Comprehensive Plan                        364
               Value Plan                                303
               Single Trip Travel Protect Gold           204
               Premier Plan                              188
               Single Trip Travel Protect Silver         173
               Annual Travel Protect Gold                100
               Annual Travel Protect Silver               86
               Individual Comprehensive Plan              74
               Single Trip Travel Protect Platinum        73
               Annual Travel Protect Platinum             53
               Spouse or Parents Comprehensive Plan       15
               Child Comprehensive Plan                    9
               Travel Cruise Protect Family                1
Name: count, dtype: int64

In [37]:
# Accepted claim counts in different product grouped into agency type
df[df['Claim'] == 'Yes'].groupby('Agency Type')['Product Name'].value_counts()

Agency Type    Product Name                        
Airlines       Bronze Plan                             210
               Annual Silver Plan                      157
               Silver Plan                             141
               Basic Plan                               23
               Annual Gold Plan                         21
               Gold Plan                                18
               Value Plan                               14
               Ticket Protector                          7
Travel Agency  2 way Comprehensive Plan                142
               Rental Vehicle Excess Insurance          86
               Cancellation Plan                        44
               Annual Travel Protect Gold               10
               Single Trip Travel Protect Gold          10
               1 way Comprehensive Plan                  9
               Comprehensive Plan                        5
               Single Trip Travel Protect Platinum       5
               Value Plan                                5
               Annual Travel Protect Silver              4
               Single Trip Travel Protect Silver         4
               Annual Travel Protect Platinum            3
               Individual Comprehensive Plan             3
               Premier Plan                              3
               Travel Cruise Protect                     2
               Spouse or Parents Comprehensive Plan      1
Name: count, dtype: int64

#**🧳 Product Performance**

###Looking at the unique product names, which products appear most frequently?

Cancellation Plan for Travel Agency & Basic Plan for Airlines

###What recommendation would you make to the marketing team?

There are lesser number of Airlines products than the travel agency products.

For the Airlines, the top 3 products are Basic plan, Bronze plan & Value plan. When the Airline insurances are ranked against accepted claims the top 3 are Bronze plan, Annual Silver plan & Silver Plan.

For travel agencies, the top 3 most popular products are cancellation plan, two way comprehensive plan & rental vehicle excess insurance plan. When the travel agency products are ranked for the accepted claims the top 3 are the same.

In [38]:
# Show the age range
df['Age'].min(), df['Age'].max(), df[df['Age'] == 118]['Age'].value_counts()

(0,
 118,
 Age
 118    984
 Name: count, dtype: int64)

In [39]:
# Show gender demographics
df['Gender'].value_counts()

Gender
M    9347
F    8872
Name: count, dtype: int64

In [40]:
# Get statistics for net sales & comision columns
print(f'minimum net sale amount: {df['Net Sales'].min()}')
print(f'maximum net sale amount: {df['Net Sales'].max()}')
print(f'minimum commision amount: {df['Commision (in value)'].min()}')
print(f'Total of net sales: {df['Net Sales'].sum():,.2f}')
print(f'Minimum duration of travel: {df['Duration'].min()}')
print(f'\n\nnumber of claims with a minimum comision amount: ')
df[df['Commision (in value)'] == 0]['Commision (in value)'].value_counts()

minimum net sale amount: -389.0
maximum net sale amount: 810.0
minimum commision amount: 0.0
Total of net sales: 2,577,495.99
Minimum duration of travel: -2


number of claims with a minimum comision amount: 


Commision (in value)
0.0    35217
Name: count, dtype: int64

# **⚠️ Data Quality**
##Did you notice anything unusual in the data (missing values, unexpected patterns)?

Yes, we can't rely on the age & gender demographics of the datasets. The age ranges from 0 to 118. There are 984 entities aged 118. Out of 63K+ entities <20K have gender information. Also there are negative values in the net sales & the duration columns. Commission values are also unreliable as >35K are 0.

##How might this affect business decisions?

The low quality data is not suitable for modeling & reporting.

This affects the performance of the marketing teams as the decisions based on Age and gender are can be erroneous due to missing data and unrealistic values.

Also it affects the finance teams to conclude business revenues due to missing & unrealistic sales & commision data.



## Which product has the highest number of claims?
Cancellation Plan

## Which has the lowest?
Travel Cruise Protect Family

## What might explain this difference?
When planning holiday tours people foresee the risk of unexpected cancellations. The insurance company highlights cancellation likelihood to sell out cancellation plan which had been for a low budget. Due to this benefit the customer tends to cancel the tour for any reason and submit claim.

Travel Cruise Protect Family product may be expensive and less number of customers are travelling on cruise. Also people plan very few cruise tours in their life so they tend to make all the arrangements.

## What recommendation would you make to the product team based on this?
To increase the pricing for cancellation plan
To introduce loyalty for Travel Cruise Protect Family & Child Comprehensive Plan.


In [41]:
# Product claim frequency
df['Product Name'].value_counts()

Product Name
Cancellation Plan                       18630
2 way Comprehensive Plan                13158
Rental Vehicle Excess Insurance          8580
Basic Plan                               5469
Bronze Plan                              4049
1 way Comprehensive Plan                 3331
Value Plan                               2715
Silver Plan                              2249
Annual Silver Plan                       1423
Ticket Protector                         1056
Travel Cruise Protect                     527
Comprehensive Plan                        364
Gold Plan                                 352
24 Protect                                247
Single Trip Travel Protect Gold           204
Premier Plan                              194
Annual Gold Plan                          194
Single Trip Travel Protect Silver         173
Annual Travel Protect Gold                100
Annual Travel Protect Silver               86
Individual Comprehensive Plan              74
Single Trip Travel Protect Platinum        73
Annual Travel Protect Platinum             53
Spouse or Parents Comprehensive Plan       15
Child Comprehensive Plan                    9
Travel Cruise Protect Family                1
Name: count, dtype: int64

In [42]:
# Average age of customers for each product type
df.groupby('Product Name')['Age'].mean().round(0)

Product Name
1 way Comprehensive Plan                37.0
2 way Comprehensive Plan                37.0
24 Protect                              49.0
Annual Gold Plan                        44.0
Annual Silver Plan                      40.0
Annual Travel Protect Gold              39.0
Annual Travel Protect Platinum          43.0
Annual Travel Protect Silver            40.0
Basic Plan                              44.0
Bronze Plan                             36.0
Cancellation Plan                       37.0
Child Comprehensive Plan                13.0
Comprehensive Plan                      63.0
Gold Plan                               40.0
Individual Comprehensive Plan           43.0
Premier Plan                            44.0
Rental Vehicle Excess Insurance         40.0
Silver Plan                             38.0
Single Trip Travel Protect Gold         40.0
Single Trip Travel Protect Platinum     41.0
Single Trip Travel Protect Silver       39.0
Spouse or Parents Comprehensive Plan    45.0
Ticket Protector                        49.0
Travel Cruise Protect                   51.0
Travel Cruise Protect Family            32.0
Value Plan                              66.0
Name: Age, dtype: float64

## What is the average age of customers for each product type?

## Which products attract older vs. younger customers?
Value Plan & Comprehensibe Plan are famous among older customers who are more than 60. Youngest range of customers hold Child comprehensive plan.

## How should the marketing team use this information when advertising these products?
In marketing campaigns the team can present what products are popular among different age categories and the reasons for those popularities.


## What are the top 5 most popular travel destinations?
## Why might these destinations be the most popular?
Because the insurance service company is based in Singapore
## Are there any destinations you're surprised to see (or not see) in the top 5?
- India & Japan are not seen in the top 5
- Some strange country names eg: space in between VIETNAM

In [43]:
# Top 10 most popular travel destinations
df['Destination'].value_counts()[:10]

Destination
SINGAPORE        13255
MALAYSIA          5930
THAILAND          5894
CHINA             4796
AUSTRALIA         3694
INDONESIA         3452
UNITED STATES     2530
PHILIPPINES       2490
HONG KONG         2411
INDIA             2251
Name: count, dtype: int64

In [44]:
# Number of different destinations
len(df['Destination'].unique())

149

In [45]:
print(list(df['Destination'].unique()))

['MALAYSIA', 'AUSTRALIA', 'ITALY', 'UNITED STATES', 'THAILAND', "KOREA, DEMOCRATIC PEOPLE'S REPUBLIC OF", 'NORWAY', 'VIET NAM', 'DENMARK', 'SINGAPORE', 'JAPAN', 'UNITED KINGDOM', 'INDONESIA', 'INDIA', 'CHINA', 'FRANCE', 'TAIWAN, PROVINCE OF CHINA', 'PHILIPPINES', 'MYANMAR', 'HONG KONG', 'KOREA, REPUBLIC OF', 'UNITED ARAB EMIRATES', 'NAMIBIA', 'NEW ZEALAND', 'COSTA RICA', 'BRUNEI DARUSSALAM', 'POLAND', 'SPAIN', 'CZECH REPUBLIC', 'GERMANY', 'SRI LANKA', 'CAMBODIA', 'AUSTRIA', 'SOUTH AFRICA', 'TANZANIA, UNITED REPUBLIC OF', "LAO PEOPLE'S DEMOCRATIC REPUBLIC", 'NEPAL', 'NETHERLANDS', 'MACAO', 'CROATIA', 'FINLAND', 'CANADA', 'TUNISIA', 'RUSSIAN FEDERATION', 'GREECE', 'BELGIUM', 'IRELAND', 'SWITZERLAND', 'CHILE', 'ISRAEL', 'BANGLADESH', 'ICELAND', 'PORTUGAL', 'ROMANIA', 'KENYA', 'GEORGIA', 'TURKEY', 'SWEDEN', 'MALDIVES', 'ESTONIA', 'SAUDI ARABIA', 'PAKISTAN', 'QATAR', 'PERU', 'LUXEMBOURG', 'MONGOLIA', 'ARGENTINA', 'CYPRUS', 'FIJI', 'BARBADOS', 'TRINIDAD AND TOBAGO', 'ETHIOPIA', 'PAPUA NEW GU

In [46]:
df[df['Destination'] == 'UKRAINE']

Agency    Agency Type Distribution Channel              Product Name  \
12201    EPX  Travel Agency               Online         Cancellation Plan   
16064    EPX  Travel Agency               Online         Cancellation Plan   
18262    EPX  Travel Agency               Online         Cancellation Plan   
18274    EPX  Travel Agency               Online         Cancellation Plan   
24042    EPX  Travel Agency               Online  2 way Comprehensive Plan   
26609    EPX  Travel Agency               Online         Cancellation Plan   
28800    EPX  Travel Agency               Online  2 way Comprehensive Plan   
48080    EPX  Travel Agency               Online  2 way Comprehensive Plan   
52229    EPX  Travel Agency               Online  2 way Comprehensive Plan   
56732    EPX  Travel Agency               Online  2 way Comprehensive Plan   

      Claim  Duration Destination  Net Sales  Commision (in value) Gender  Age  
12201    No        12     UKRAINE       80.0                   0.0    NaN   36  
16064    No        11     UKRAINE       80.0                   0.0    NaN   36  
18262    No       120     UKRAINE       75.0                   0.0    NaN   36  
18274    No        39     UKRAINE       12.0                   0.0    NaN   36  
24042    No        34     UKRAINE       58.0                   0.0    NaN   36  
26609    No        62     UKRAINE       80.0                   0.0    NaN   36  
28800    No       104     UKRAINE       74.0                   0.0    NaN   36  
48080    No       114     UKRAINE      116.0                   0.0    NaN   36  
52229    No        19     UKRAINE      104.0                   0.0    NaN   36  
56732    No        45     UKRAINE      112.0                   0.0    NaN   36

## Analyse negative sales percentages

In [47]:
negative_sales_percentage = (df[df['Net Sales'].lt(0)].shape[0] / df.shape[0]) * 100
print(f'Negative sales percentage is {negative_sales_percentage:.2f}%')

Negative sales percentage is 1.07%


In [48]:
df[(df['Net Sales'].lt(0)) & (df['Claim'] == 'Yes')].shape[0], df[(df['Net Sales'].lt(0)) & (df['Claim'] == 'No')].shape[0]

(3, 675)

## What possible explanations are there for these negative values?
- This can be a data entry error or calculation error
- A large number of negative net sale values are rejected claims

# What would you recommend to the Sales team leader?
- Definition for the Net sales metric
- When calculating net sales define a default value

## Compare claims by age groups

In [52]:
# Define age ranges for the age distribution
df['Age cuts'] = pd.cut(df['Age'], 10)
df['Age cuts'].unique()

[(70.8, 82.6], (23.6, 35.4], (35.4, 47.2], (106.2, 118.0], (47.2, 59.0], (59.0, 70.8], (11.8, 23.6], (82.6, 94.4], (-0.118, 11.8]]
Categories (10, interval[float64, right]): [(-0.118, 11.8] < (11.8, 23.6] < (23.6, 35.4] <
                                            (35.4, 47.2] ... (70.8, 82.6] < (82.6, 94.4] <
                                            (94.4, 106.2] < (106.2, 118.0]]

In [53]:
# Define age groups manually
df['Age Groups'] = df['Age'].map(
    lambda x: '0+' if x < 10 else '10+' if x<20 else '20+' if x < 30 else '30+' if x < 40 else '40+' if x < 50
    else '50+' if x < 60 else '60+' if x < 70 else '70+' if x < 80 else '80+' if x < 90 else '90+' if x < 100 else '100+' if x < 110 else '110+')

In [54]:
# Distribution in different age groups
df['Age Groups'].value_counts()

Age Groups
30+     36541
40+      9430
20+      7515
50+      5406
60+      2430
110+      984
70+       643
10+       220
80+       124
0+         33
Name: count, dtype: int64

## Do older customers make more or fewer claims than younger customers?
- More than 50% of the claims are from people aged 30-39

In [55]:
# Derive claim rejection rates for age groups
(df[df['Claim'] == 'No']['Age Groups'].value_counts() / df['Age Groups'].value_counts()).round(2)

Age Groups
30+     0.99
40+     0.99
20+     0.97
50+     0.98
60+     0.98
110+    0.99
70+     0.98
10+     0.97
80+     0.98
0+      1.00
Name: count, dtype: float64

## Which age group is the riskiest for the insurance company?
- 0-10 age group is the riskiest with a 100% rejection rate

## Based on this analysis, should the company:
## Charge different prices for different age groups?
- No, the data quality should be improved first

## Target marketing at specific age groups?
- No, the data seems to have a serious imbalance which have to be questioned from the data team

## What are the ethical implications of age-based pricing?
- Market segmentation based on age will promote equity based on age but it will create age discrimination. Also it can affect consumer protection & have a exploitation risk.